### PR TITLE
FEATURE: Allow admin to change timestamp of topic.

### PR DESCRIPTION
**Pull request by tgxworld**
meta https://meta.discourse.org/t/update-changing-timestamp-through-ui/29464

### CODE REVIEW

File: app/assets/javascripts/discourse/controllers/change-timestamp.js.es6 [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  time: '',
+
+  buttonTitle: function() {
+    return this.get('saving') ? I18n.t('saving') : I18n.t('topic.change_timestamp.action');
+  }.property('saving'),
+
+  buttonDisabled: function() {
+    if (this.get('saving') || this.get('validTimestamp')) return true;
+    return (this.blank('date'));
+  }.property('saving', 'validTimestamp'),
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/controllers/change-timestamp.js.es6]: Shouldn't this also observe "`date`"?

File: app/assets/javascripts/discourse/controllers/change-timestamp.js.es6 [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  createdAt: function() {
+    return this.get('date') + ' ' + this.get('time');
+  },
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/controllers/change-timestamp.js.es6]: missing a `}.property("date", "time")`?

File: app/assets/javascripts/discourse/controllers/change-timestamp.js.es6 [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      this.set('saving', true);
+      const self = this;
+
+      Discourse.Topic.changeTimestamp(
+        this.get('topicController.model.id'),
+        moment(this.createdAt()).unix()
+      ).then(function() {
+        self.send('closeModal');
+        self.setProperties({ date: '', time: '', saving: false });
+      }, function() {
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/controllers/change-timestamp.js.es6]: We prefer the `catch` syntax for failing promises.

**Comment by ZogStriP** [comment]
I like that. Just a couple of notes though ;)

**Comment by tgxworld** [comment]
@ZogStriP Updated to remove some Ember 1.12 deprecations and also to use `ember-computed-decorators` :smile:  Thanks for reviewing!

**Comment by ZogStriP** [comment]
:+1:

**Comment by dylanh724** [comment]
ooh is this is in the latest update?

**Pull request closed without merging**